File: src/AddAPearl.Domain/Address.cs
using AddAPearl.Core;

namespace AddAPearl.Domain
{
    public class Address: IAddress
    {
        public int AddressId { get; set; }
        
        public string AddressLine1 { get; set; }
        
        public string AddressLine2 { get; set; }
        
        public string AddressLine3 { get; set; }
        
        public string City { get; set; }
        
        public string State { get; set; }
        
        public string ZipCode { get; set; }
    }
}

File: AddAPearl.Web/src/app/shared/models/address.ts
/* Defines the address entity */
export interface IAddress {
	addressId: number;
	addressLine1: string;
	addressLine2: string;
	addressLine3: string;
	city: string;
	state: string;
	zipCode: string;
}

export class Address implements IAddress {
	constructor(
		public addressId: number,
		public addressLine1: string,
		public addressLine2: string,
		public addressLine3: string,
		public city: string,
		public state: string,
		public zipCode: string,
	) {
		// Any other stuff to do during construction...
	}
}
